kotlinx-coroutines-core/common/src/EventLoop.common.kt

```diff
@@ -123,6 +123,10 @@ internal abstract class EventLoop : CoroutineDispatcher() {
     open fun shutdown() {}
 }
 
+/*
+ * This is marked with `ThreadLocal` on purpose: the loop itself is supposed to be thread-local,
+ * but `CommonThreadLocal` is not supported on K/N properly, so we emulate it with this annotation.
+ */
```

Comment:
`CommonThreadLocal` is quite a leaky abstraction then. Maybe we should fix it instead? Here's my attempt to do it: https://gist.github.com/dkhalanskyjb/7cc2d6c752d97217c6c288304253ba11

Reply:
In general, I prefer to extract abstraction (if we are talking about internal API of course) only when it has at least two uses because otherwise, it's much easier to either over-abstract or write too much code for no good.

But it is already here, so I've pushed the change

Reply:
Agreed: if there was no `CommonThreadLocal`, I wouldn't propose one.

Reply:
BTW, https://stackoverflow.com/a/50329788